src/bvh/optimization.rs

```diff
@@ -354,19 +363,13 @@ impl<T: BHValue, const D: usize> Bvh<T, D> {
             self.nodes[node_index] = self.nodes[end];
             let parent_index = self.nodes[node_index].parent();
 
-            if let BvhNode::Leaf { .. } = self.nodes[parent_index] {
-                self.nodes.truncate(end);
-                return;
-            }
             let parent = self.nodes[parent_index];
+            assert!(!parent.is_leaf());
             let moved_left = parent.child_l() == end;
-            if !moved_left && parent.child_r() != end {
-                self.nodes.truncate(end);
-                return;
-            }
```

Comment:
Why is it safe to take this out? I admit my understanding of the implementation is a bit rusty at this point.

Reply:
If the **parent** of a node is a **leaf** node, the BVH is inconsistent. 75%+ of the code will panic on an inconsistent BVH, as it should. It shouldn't be possible for the BVH to become inconsistent. Trying to handle cases that should be impossible in a small minority of places isn't a solution to bugs that make the BVH inconsistent. The fuzzer is supposed to catch all such bugs in advance ;)